File: app/src/main/java/com/github/overpass/gather/screen/meeting/chat/attachments/PhotosActivity.java
package com.github.overpass.gather.screen.meeting.chat.attachments;

import android.content.Context;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.view.MenuItem;

import androidx.annotation.NonNull;
import androidx.appcompat.app.ActionBar;
import androidx.core.content.ContextCompat;
import androidx.lifecycle.ViewModelProviders;

import com.github.overpass.gather.R;
import com.github.overpass.gather.model.commons.FragmentUtils;
import com.github.overpass.gather.screen.base.BaseActivity;
import com.github.overpass.gather.screen.dialog.PickImageDialogFragment;

import static com.github.overpass.gather.model.commons.Constants.MEETING_ID_KEY;

public class PhotosActivity extends BaseActivity<GeneralPhotoViewModel>
        implements PickImageDialogFragment.OnClickListener {

    @Override
    protected int getLayoutRes() {
        return R.layout.activity_photos;
    }

    @Override
    protected GeneralPhotoViewModel createViewModel() {
        return ViewModelProviders.of(this).get(GeneralPhotoViewModel.class);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (savedInstanceState == null) {
            FragmentUtils.replace(getSupportFragmentManager(), R.id.flPhotosContainer,
                    PhotosFragment.newInstance(getMeetingId()), false);
        }
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            finish();
            return true;
        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    protected void onActionBar(@NonNull ActionBar actionBar) {
        int primaryColor = ContextCompat.getColor(this, R.color.colorPrimary);
        actionBar.setBackgroundDrawable(new ColorDrawable(primaryColor));
        actionBar.setTitle(R.string.attachments);
    }

    @Override
    public void onGallery() {
        viewModel.onGallery();
    }

    @Override
    public void onCamera() {
        viewModel.onCamera();
    }

    private String getMeetingId() {
        String meetingId = "-1";
        Bundle extras = getIntent().getExtras();
        if (extras != null) {
            meetingId = extras.getString(MEETING_ID_KEY);
        }
        return meetingId;
    }

    public static void start(Context context, String meetingId) {
        start(context, PhotosActivity.class, MEETING_ID_KEY, meetingId);
    }
}

File: app/src/main/java/com/github/overpass/gather/screen/meeting/chat/attachments/PhotosActivity.kt
package com.github.overpass.gather.screen.meeting.chat.attachments

import android.content.Context
import android.graphics.drawable.ColorDrawable
import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.ActionBar
import androidx.core.content.ContextCompat
import com.github.overpass.gather.App.Companion.appComponentManager
import com.github.overpass.gather.R
import com.github.overpass.gather.di.meeting.chat.attachments.AttachmentsComponent
import com.github.overpass.gather.di.meeting.chat.attachments.AttachmentsComponentManager
import com.github.overpass.gather.model.commons.Constants.MEETING_ID_KEY
import com.github.overpass.gather.model.commons.FragmentUtils
import com.github.overpass.gather.model.commons.getStringExtra
import com.github.overpass.gather.screen.base.BaseActivity
import com.github.overpass.gather.screen.dialog.PickImageDialogFragment

class PhotosActivity : BaseActivity<GeneralPhotoViewModel, AttachmentsComponent>(),
        PickImageDialogFragment.OnClickListener {

    override val componentManager: AttachmentsComponentManager =
            appComponentManager.getAttachmentsComponentManager()

    override fun createComponent(): AttachmentsComponent = componentManager.getOrCreate(Unit)

    override fun onComponentCreated(component: AttachmentsComponent) {
        component.inject(this)
    }

    override val layoutRes: Int = R.layout.activity_photos

    override fun createViewModel(): GeneralPhotoViewModel {
        return viewModelProvider.get(GeneralPhotoViewModel::class.java)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (savedInstanceState == null) {
            FragmentUtils.replace(supportFragmentManager, R.id.flPhotosContainer,
                    PhotosFragment.newInstance(getStringExtra(MEETING_ID_KEY)), false)
        }
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            finish()
            return true
        }
        return super.onOptionsItemSelected(item)
    }

    override fun onActionBar(actionBar: ActionBar) {
        val primaryColor = ContextCompat.getColor(this, R.color.colorPrimary)
        actionBar.setBackgroundDrawable(ColorDrawable(primaryColor))
        actionBar.setTitle(R.string.attachments)
    }

    override fun onGallery() {
        viewModel.onGallery()
    }

    override fun onCamera() {
        viewModel.onCamera()
    }

    companion object {

        fun start(context: Context, meetingId: String) {
            start(context, PhotosActivity::class.java, MEETING_ID_KEY, meetingId)
        }
    }
}
